.ci/ci-fast-return.sh

```diff
@@ -113,6 +121,63 @@ check_not_matches() {
 	egrep -v $@ <<< "$filenames" || true
 }
 
+# Check if the specified label is set on a PR.
+#
+# Returns on stdout as string:
+#
+#  0 - Label not found.
+#  1 - Label found.
+is_label_set() {
+	local repo="${1:-}"
+	local pr="${2:-}"
+	local label="${3:-}"
+
+	if [ -z "$repo" ]; then
+		local_info "BUG: No repo specified"
+		echo "0"
+		return 0
+	fi
+
+	if [ -z "$pr" ]; then
+		local_info "BUG: No PR specified"
+		echo "0"
+		return 0
+	fi
+
+	if [ -z "$label" ]; then
+		local_info "BUG: No label to check specified"
+		echo "0"
+		return 0
+	fi
+
+	local_info "Checking labels for PR ${repo}/${pr}"
+
+	# Pull the label list for the PR
+	# Ideally we'd use a github auth token here so we don't get rate limited, but to do that we would
+	# have to expose the token into the CI scripts, which is then potentially a security hole.
+	local json=$(curl -sL "https://api.github.com/repos/${repo}/issues/${pr}/labels")
+
+	install_jq
+
+	# Pull the label list out
+	local labels=$(jq '.[].name' <<< $json)
+
+	# Check if we have the forcing label set
+	for x in $labels; do
+		# Strip off any surrounding '"'s
+		y=$(sed 's/"//g' <<< $x)
+
+		if [ "$y" == "$label" ]; then
+			echo "1"
+			return 0
```

Comment:
I must admit that looking at and checking whatever is printed to stdout rather than whatever is returned is confusing.

Reply:
Also agreed. I think this is partly due to the constraints of shunit2 / `set -e`.